third_party/amd/backend/compiler.py

```diff
@@ -46,6 +46,16 @@ class HIPOptions:
     max_num_imprecise_acc_default: int = 0
     backend_name: str = 'hip'
 
+    # the following option provides hints to the AMDGPU backend
+    # regarding instruction scheduling for all `tt.dot` operations
+    # in a kernel. The default value (i.e., `0`) preserves
+    # the default instruction scheduling of the AMDGPU backend
+    # which aims at maximizing occupancy.
+    # The option is experimental and may change at
+    # any time regarding its semantics and/or may be gone
+    # entirely anytime.
+    instruction_sched_variant: int = 0
```

Comment:
Can we accept a string here like `iglp0`, `iglp1`, `schedbarrier`, etc. Easier to play with this way. You can translate these strings to internal numeric representations in the pass. Also allows us to renumer them easily.

Reply:
Done.

I am not sure how it is going to scale in the future b/c I assume that we will have more than on `schedbarrier` variant. But let's see. We can revert it in the future if necessary

Reply:
This is not done? Did you forget to push out a commit or something? Also need to update the above comment accordingly.